### PR TITLE
Fixing links and adding missing content

### PR DESCRIPTION
**Pull request by ardalis**
## Summary

Added 2 missing sections to Chapter 2.
Fixed a bunch of internal links required by the PDF generator.

### CODE REVIEW

File: docs/architecture/cloud-native/other-deployment-options.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+description: Architecting Cloud Native .NET Apps for Azure | Other Deployment Options
+ms.date: 06/30/2019
+---
+# Other Deployment Options
+
+In addition to deploying to AKS, you can also deploy containers to Azure App Service for Containers and Azure Container Instances. Simple production applications that don't require orchestration are well-suited to Azure App Service for Containers. Azure Container Instances are best used for testing scenarios.
+
+## App Service for Containers
+
+**Note**: Do we need to cover deployment to App Service for Containers here or is that covered elsewhere?
```
Comment (rpetrusha) [inline docs/architecture/cloud-native/other-deployment-options.md]: Can this be removed?

File: docs/architecture/cloud-native/other-deployment-options.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+## Azure Container Instances
+
+To deploy to [Azure Container Instances (ACI)](https://docs.microsoft.com/azure/container-instances/), you need to have configured an Azure Container Registry (ACR) and credentials for accessing it. You must also have previously pushed your container image to the registry, so it's available to pull into ACI. You can work with ACI using the Azure CLI or through the portal. Azure Container Registries make it easy to deploy individual container instances to ACI directly from within the registry, as shown in Figure 2-x.
```
Comment (rpetrusha) [inline docs/architecture/cloud-native/other-deployment-options.md]:
```suggestion
To deploy to [Azure Container Instances (ACI)](/azure/container-instances/), you need to have configured an Azure Container Registry (ACR) and credentials for accessing it. You must also have previously pushed your container image to the registry, so it's available to pull into ACI. You can work with ACI using the Azure CLI or through the portal. Azure Container Registries make it easy to deploy individual container instances to ACI directly from within the registry, as shown in Figure 2-x.
```

File: docs/architecture/cloud-native/other-deployment-options.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+![Azure Container Registry Run Instance](./media/acr-runinstance-contextmenu.png)
+**Figure 2-X**. Azure Container Registry Run Instance
+
+Creating a container instance from the registry just requires specifying the usual Azure settings (name, subscription, resource group, and location) as well as how much memory to allocate to the container and which port it should listen on. Figure 2-X shows the Azure Portal form used to create the new instance.
+
+![ACR Run Instance Create container instance](./media/acr-create-deeplink.png)
+**Figure 2-X**. ACR Run Instance Create container instance
+
+Once the deployment completes, you can find the newly-deployed container's IP address and communicate with it over the port you specified.
```
Comment (rpetrusha) [inline docs/architecture/cloud-native/other-deployment-options.md]:
```suggestion
Once the deployment completes, you can find the newly deployed container's IP address and communicate with it over the port you specified.
```

**Review by rpetrusha — CHANGES_REQUESTED** [review]
I've left some suggestions and comments, @ardalis. The figures also need to be numbered appropriately.

File: docs/architecture/cloud-native/deploy-containers-azure.md [hunk 4]
```diff
@@ -5,7 +5,108 @@ ms.date: 06/30/2019
...
+
+## Azure Container Registry
+
+Azure Container Registry lets you build, store, and manage images for all of your container deployments. You can [create a container registry using the Azure Portal](https://docs.microsoft.com/azure/container-registry/container-registry-get-started-portal) or [using the Azure CLI](https://docs.microsoft.com/azure/container-registry/container-registry-get-started-azure-cli) or [PowerShell tools](https://docs.microsoft.com/azure/container-registry/container-registry-get-started-powershell). Creating a new container registry just requires an Azure subscription, a resource group, and a unique name. Figure 2-x shows the basic options for creating a registry, which will be hosted at *registryname*.azurecr.io.
+
+![Create container registry](media/create-container-registry.png)
+**Figure 2-x**. Create container registry
```
[the remark this answers is not in the document]
Comment (ardalis) [reply docs/architecture/cloud-native/deploy-containers-azure.md]: Will do a separate PR with that.

File: docs/architecture/cloud-native/other-deployment-options.md (hunk 1 above)
Comment (ardalis) [reply docs/architecture/cloud-native/other-deployment-options.md]: Yes, thanks.

**Comment by rpetrusha** [comment]
@ardalis, can you address the merge conflict?

**Comment by ardalis** [comment]
Can we merge this and I'll do a pass to fix all the figure numbers? Thanks.

**Comment by rpetrusha** [comment]
Yes, I'll approve and merge now, @dotnet/nsboard-xamarin

**Review by rpetrusha — APPROVED** [review]

**Pull request merged**